api/src/main/java/ca/bc/gov/educ/eas/api/service/v1/events/EventHandlerService.java

```diff
@@ -38,118 +42,127 @@
     }
 
-    log.trace(EVENT_PAYLOAD, event);
-    val optionalStudentEntity = assessmentStudentRepository.findByAssessmentEntity_AssessmentIDAndStudentID(UUID.fromString(student.getAssessmentID()), UUID.fromString(student.getStudentID()));
-    if (optionalStudentEntity.isPresent()) {
-      AssessmentStudent structStud = assessmentStudentMapper.toStructure(optionalStudentEntity.get()); // need to convert to structure MANDATORY otherwise jackson will break.
-      event.setEventPayload(JsonUtil.getJsonStringFromObject(structStud));
-      event.setEventOutcome(EventOutcome.STUDENT_REGISTRATION_FOUND);
-    } else {
-      event.setEventOutcome(EventOutcome.STUDENT_REGISTRATION_NOT_FOUND);
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public void handlePublishStudentRegistrationEvent(final Event event) throws JsonProcessingException {
+        if (event.getEventOutcome() == EventOutcome.STUDENT_REGISTRATION_EVENT_READ) {
+            final AssessmentStudent sagaData = JsonUtil.getJsonObjectFromString(AssessmentStudent.class, event.getEventPayload());
+            final var sagaOptional = sagaService.findByAssessmentStudentIDAndSagaNameAndStatusNot(UUID.fromString(sagaData.getAssessmentStudentID()), SagaEnum.PUBLISH_STUDENT_REGISTRATION.toString(), SagaStatusEnum.COMPLETED.toString());
+            if (sagaOptional.isPresent()) { // possible duplicate message.
+                log.trace("Execution is not required for this message returning EVENT is :: {}", event);
+                return;
+            }
+            val saga = this.studentRegistrationOrchestrator.createSaga(event.getEventPayload(), sagaData.getUpdateUser(), UUID.fromString(sagaData.getAssessmentStudentID()));
+            log.debug("Starting student processing orchestrator :: {}", saga);
+            this.studentRegistrationOrchestrator.startSaga(saga);
+        }
     }
-    val studentEvent = createEventRecord(event);
-    return createResponseEvent(studentEvent);
-  }
 
-  private EasEventEntity createEventRecord(Event event) {
-    return EasEventEntity.builder()
-        .createDate(LocalDateTime.now())
-        .updateDate(LocalDateTime.now())
-        .createUser(event.getEventType().toString()) //need to discuss what to put here.
-        .updateUser(event.getEventType().toString())
-        .eventPayloadBytes(event.getEventPayload().getBytes(StandardCharsets.UTF_8))
-        .eventType(event.getEventType().toString())
-        .sagaId(event.getSagaId())
-        .eventStatus(MESSAGE_PUBLISHED.toString())
-        .eventOutcome(event.getEventOutcome().toString())
-        .replyChannel(event.getReplyTo())
-        .build();
-  }
+    private EasEventEntity createEventRecord(Event event) {
+        return EasEventEntity.builder()
+                .createDate(LocalDateTime.now())
+                .updateDate(LocalDateTime.now())
+                .createUser(event.getEventType().toString()) //need to discuss what to put here.
```

Comment:
EAS_API in the application properties can be used for update/create user.

Reply:
Changes applied.